src/checkL2Version.ts

```diff
@@ -0,0 +1,73 @@
+import { exec } from "child_process";
+import * as vscode from "vscode";
+import { getShowLama2Term } from "./utils";
+
+const MIN_VERSION_TO_CHECK = "1.5.2";
+const LAMA2_TERM_NAME = "AutoLama2";
+const UPDATE_MSG = "Support for environment variables.";
+
+export function checkL2Version() {
+  try {
+    getL2Version((l2Version) => {
+      // Check if version is below MIN_VERSION_TO_CHECK
+      if (compareL2Versions(l2Version, MIN_VERSION_TO_CHECK) < 0) {
+        showUpdateWarning();
+      }
+    });
+  } catch (e: any) {
+    console.log("Problem while checking for version -> ", e);
+  }
+}
+
+function getL2Version(callback: (version: string) => void) {
+  exec("l2 --version", (error, stdout, stderr) => {
+    const versionOutput = stdout.trim();
+    const versionMatch = versionOutput.match(/v(\d+\.\d+\.\d+)/);
+    if (versionMatch && versionMatch.length === 2) {
+      const l2Version = versionMatch[1];
+      callback(l2Version);
+    }
+  });
+}
+
+function compareL2Versions(currentVersion: string, latestVersion: string) {
```

Comment:
I recommend using a library recommended in this [SO Answer](https://stackoverflow.com/a/30256597). Our semver definition is simple now, but there are more changes possible. Best to simply use standardized library for it.

Should be as simple as:

```
semver.gte('3.4.8', '3.4.7') //true
```

Reply:
If we're using this library, then ignore the previous comment about regex*.

Reply:
Yeah using this